### PR TITLE
:seedling: test: add `go/v3-alpha` e2e tests

### PR DESCRIPTION
**Pull request by estroz**
This PR adds e2e tests for go/v3-alpha and refactors test code such that it is easier to add other plugin tests going forward.

- test/e2e/utils: add util to remove tool binaries from path
- test/e2e/v3: refactor generation code so all plugins can be tested
- *: clean up scripts

Pulls refactoring/cleanup from #1644

### CODE REVIEW

**Comment by estroz** [comment]
Relates to #1746.

File: test/e2e/v3/e2e_suite_test.go
Comment (camilamacedo86) [inline test/e2e/v3/e2e_suite_test.go:1]: Hi @esric, 

would be possible take advantage that you are working on with and keep the suite_test.go just to trigger the tests? The suite_test should not have the tests actually. Could we do here something such as follows? 

- suite_test.go : env test setup, prequirements installs
- v2_plugin_cluster_test.go: for the v2 plugin tests
- v3_plugin_cluster_test.go: for the v3 plugin tests

Comment (estroz) [reply test/e2e/v3/e2e_suite_test.go:1]: Sure I can rename test files.

Comment (camilamacedo86) [reply test/e2e/v3/e2e_suite_test.go:1]: I have a todo note for a while to work on in its cleanups and try to do as we did with SDK tests. So, if you be able to do that will be great. However, I do not want blocker your pr because of the cleanup.

Comment (estroz) [reply test/e2e/v3/e2e_suite_test.go]: Done.

**Comment by estroz** [comment]
Blocked by #1746

**Comment by christopherhein** [comment]
> Blocked by #1746

In the interim can we use an alpha release like https://github.com/kubernetes-sigs/kubebuilder/pull/1790/files#diff-e33c8c9eef8145c9616af064f8df83cdc5a36371b5884f39e279fc1dc87535a7R40 I don't want to duplicate fixing up all the tests for 0.7.0 if I can help it. See https://github.com/kubernetes-sigs/kubebuilder/pull/1790/files#r518356945 for @Adirio and @camilamacedo86's feedback

**Comment by estroz** [comment]
@christopherhein agreed, but for tests to pass https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/127 needs to be merged first.

**Comment by estroz** [comment]
Closing, see https://github.com/kubernetes-sigs/kubebuilder/pull/1787#issuecomment-724373967

**Comment by estroz** [comment]
Reopening to add tests so reverted PR's can be re-submitted.

**Comment by estroz** [comment]
Blocked by #1806, #1807

**Comment by Adirio** [comment]
Seems to be some issue with authorization required for leader election:
```
E1110 04:32:36.089670       1 leaderelection.go:325] error initially creating leader election record: leases.coordination.k8s.io is forbidden: User "system:serviceaccount:e2e-ewba-system:default" cannot create resource "leases" in API group "coordination.k8s.io" in the namespace "e2e-ewba-system"
E1110 04:32:39.545701       1 leaderelection.go:321] error retrieving resource lock e2e-ewba-system/a3b38f53.example.comewba: leases.coordination.k8s.io "a3b38f53.example.comewba" is forbidden: User "system:serviceaccount:e2e-ewba-system:default" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "e2e-ewba-system"
```

**Comment by estroz** [comment]
@Adirio fixed in #1809

File: test/e2e/v3/plugin_cluster_test.go
Comment (estroz) [inline test/e2e/v3/plugin_cluster_test.go:212]: Metrics are now being used to confirm a successful reconcile since the "Successfully Reconcile" log message was [removed](https://github.com/kubernetes-sigs/controller-runtime/pull/1096).

**Comment by Adirio** [comment]
Once #1809 (linked above) gets merged, this may need a rebase.

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

**Pull request merged**
